src/hotspot/share/memory/allocation.hpp

```diff
@@ -315,7 +315,8 @@ class MetaspaceObj {
   f(Annotations) \
   f(MethodCounters) \
   f(SharedClassPathEntry) \
-  f(RecordComponent)
+  f(RecordComponent) \
+  f(AbstractClass)
```

Comment:
Yes, this is a bit inconsistent.  I like your suggestion for an improvement.  I was going to try to do this in this patch but the MetaspaceObj::Type is used for the report_metadata_oom event, so it's still needed in metaspace::allocate.  Which is unfortunate because I always get these two MetadataType and MetaspaceObj::Type things confused.